File: src/main/java/net/matees/arcade/mobrush/settings/EnablePeacefulMobs.java
package net.matees.arcade.mobrush.settings;

import org.bukkit.Material;

import net.matees.settings.BooleanSetting;

public class EnablePeacefulMobs extends BooleanSetting {

    private Boolean setting = true;
    private static final EnablePeacefulMobs INSTANCE = new EnablePeacefulMobs();

    private EnablePeacefulMobs() {
    }

    public static EnablePeacefulMobs getInstance() {
        return INSTANCE;
    }

    @Override
    public String getName() {
        return "Enable Peaceful Mobs";
    }

    @Override
    public String getDescription() {
        return "Allows spawning of peaceful mobs";
    }

    @Override
    public Boolean getSetting() {
        return setting;
    }

    @Override
    public void setSetting(Boolean setting) {
        this.setting = setting;
    }

    @Override
    public Material getMenuItemMaterial() {
        return Material.SHEEP_SPAWN_EGG;
    }

    @Override
    public int getMenuItemSlot() {
        return 14;
    }

}

File: src/main/java/net/matees/arcade/mobrush/settings/EnablePeacefulMobs.kt
package net.matees.arcade.mobrush.settings

import net.matees.settings.AbstractBooleanSetting
import org.bukkit.Material

class EnablePeacefulMobs private constructor() : AbstractBooleanSetting() {
    override var setting: Boolean? = true

    override val name: String
        get() = "Enable Peaceful Mobs"

    override val description: String
        get() = "Allows spawning of peaceful mobs"

    override val menuItemMaterial: Material
        get() = Material.SHEEP_SPAWN_EGG

    override val menuItemSlot: Int
        get() = 14

    companion object {
        val instance: EnablePeacefulMobs = EnablePeacefulMobs()
    }
}
